src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2724,6 +2724,12 @@ internal static uint GetValEscape(BoundExpression expr, uint scopeOfTheContainin
                     var switchExpr = (BoundSwitchExpression)expr;
                     return GetValEscape(switchExpr.SwitchArms.SelectAsArray(a => a.Value), scopeOfTheContainingExpression);
 
+                case BoundKind.WithExpression:
```

Comment:
>case BoundKind.WithExpression: [](start = 16, length = 30)

It might be convenient to place this case next to ```BoundKind.ObjectCreationExpression:``` since they both have to deal with an initializer. Same for the other function getting a similar case.